### PR TITLE
Fixed a bug of gradient computation in rotation_matrix_to_angle_axis

### PR DESCRIPTION
**Pull request by sh8**
#### Changes
As I pointed out in [this issue](https://github.com/kornia/kornia/issues/1267), there is a bug or numerical instability issue of gradient computation in the `rotation_matrix_to_angle_axis` function. I fixed the problem by implementing a direct conversion that bypasses quaternion conversions. As far as I tested on SMPLify, it works well. However, we may need more test cases to make sure this fix solves the issue correctly. Also, I'm suspecting the root cause of this issue is in quaternion conversion functions. I will investigate it when I have time.

Fixes # (issue)
Implemented a fixed version of the `rotation_matrix_to_angle_axis` function.

#### Type of change

- [x ] 🐞 Bug fix (non-breaking change which fixes an issue)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: kornia/geometry/conversions.py [hunk 1]
```diff
@@ -321,8 +321,21 @@ def rotation_matrix_to_angle_axis(rotation_matrix: torch.Tensor) -> torch.Tensor
 
     if not rotation_matrix.shape[-2:] == (3, 3):
         raise ValueError(f"Input size must be a (*, 3, 3) tensor. Got {rotation_matrix.shape}")
-    quaternion: torch.Tensor = rotation_matrix_to_quaternion(rotation_matrix, order=QuaternionCoeffOrder.WXYZ)
-    return quaternion_to_angle_axis(quaternion, order=QuaternionCoeffOrder.WXYZ)
+
+    axis = torch.zeros((rotation_matrix.shape[0], 3), device=rotation_matrix.device)
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:325]:
```suggestion
    axis = torch.empty((rotation_matrix.shape[0], 3), device=rotation_matrix.device, dtype=rotation_matrix.dry-es)
```

File: kornia/geometry/conversions.py [hunk 2]
```diff
@@ -321,8 +321,21 @@ def rotation_matrix_to_angle_axis(rotation_matrix: torch.Tensor) -> torch.Tensor
...
+    axis[:, 1] = rotation_matrix[:, 0, 2] - rotation_matrix[:, 2, 0]
+    axis[:, 2] = rotation_matrix[:, 1, 0] - rotation_matrix[:, 0, 1]
+
+    # add epsilon for numerical stability
+    r = torch.norm(axis, dim=1).unsqueeze(1) + eps
+    t = rotation_matrix[:, 0, 0] + rotation_matrix[:, 1, 1] + rotation_matrix[:, 2, 2]
+    t = t.unsqueeze(1)
+    # use atan2 instead of torch.acos((t - 1)/2) for numerical stability
+    theta = torch.atan2(r, t - 1)
+    axis = axis / r
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:336]:
```suggestion
    axis = axis / (r + eps)
```
Not sure about this one here

File: kornia/geometry/conversions.py [hunk 3]
```diff
@@ -321,8 +321,21 @@ def rotation_matrix_to_angle_axis(rotation_matrix: torch.Tensor) -> torch.Tensor
 
     if not rotation_matrix.shape[-2:] == (3, 3):
         raise ValueError(f"Input size must be a (*, 3, 3) tensor. Got {rotation_matrix.shape}")
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:323]: Probably we should update this docs and checking len == 3 as -er what says above

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: kornia/geometry/conversions.py (hunk 2 above)
Comment (sh8) [reply kornia/geometry/conversions.py]: :+1:

File: kornia/geometry/conversions.py (hunk 3 above)
Comment (sh8) [reply kornia/geometry/conversions.py:323]: :+1:

File: kornia/geometry/conversions.py (hunk 1 above)
Comment (sh8) [reply kornia/geometry/conversions.py]: :+1:

**Comment by sh8** [comment]
thank you for the review! I fixed the code based on your comments.

File: kornia/geometry/conversions.py [hunk 4]
```diff
@@ -319,10 +319,23 @@ def rotation_matrix_to_angle_axis(rotation_matrix: torch.Tensor) -> torch.Tensor
     if not isinstance(rotation_matrix, torch.Tensor):
         raise TypeError(f"Input type is not a torch.Tensor. Got {type(rotation_matrix)}")
 
-    if not rotation_matrix.shape[-2:] == (3, 3):
+    if not len(rotation_matrix.shape) == 3 or not rotation_matrix.shape[-2:] == (3, 3):
         raise ValueError(f"Input size must be a (*, 3, 3) tensor. Got {rotation_matrix.shape}")
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:323]:
```suggestion
        raise ValueError(f"Input size must be a (N, 3, 3) tensor. Got {rotation_matrix.shape}")
```

**Comment by sh8** [comment]
Ah... Let me take a look at the error tomorrow!

**Comment by edgarriba** [comment]
@sh8 did you check this ?
